src/metldata/builtin_transformations/common/assumptions.py

```diff
@@ -0,0 +1,160 @@
+# limitations under the License.
+
+"Assumptions utilized in more than one transform types"
+
+from schemapack.spec.schemapack import SchemaPack
+
+from metldata.builtin_transformations.add_content_properties.path import (
+    resolve_schema_object_path,
+)
+from metldata.builtin_transformations.common.instruction import InstructionProtocol
+from metldata.builtin_transformations.common.path.path import RelationPath
+from metldata.builtin_transformations.common.path.path_elements import (
+    RelationPathElement,
+    RelationPathElementType,
+)
+from metldata.transform.exceptions import ModelAssumptionError, MultiplicityError
+
+
+def assert_path_classes_and_relations_exist(*, model: SchemaPack, path: RelationPath):
+    """Make sure that all classes and relations defined in the provided path exist in
+    the provided model.
+
+    Raises:
+        ModelAssumptionError:
+            if the model does not fulfill the assumptions.
+    """
+    for path_element in path.elements:
+        _check_class_exists(model, path_element.source)
+        _check_class_exists(model, path_element.target)
+
+        if path_element.type_ == RelationPathElementType.ACTIVE:
+            _check_relation_exists(model, path_element.source, path_element.property)
+
+        if path_element.type_ == RelationPathElementType.PASSIVE:
+            _check_relation_exists(model, path_element.target, path_element.property)
+
+
+def _check_class_exists(model: SchemaPack, class_name: str) -> None:
+    """Check if a class exists in the model and raise an error if not"""
+    if class_name not in model.classes:
+        raise ModelAssumptionError(f"Class {class_name} not found in model.")
+
+
+def _check_relation_exists(model: SchemaPack, class_name: str, relation: str):
+    """Check if a relation exists in a class and raise an error if not"""
+    if relation not in model.classes[class_name].relations:
+        raise ModelAssumptionError(
+            f"Relation property {
+                relation} not found in class {class_name}."
+        )
+
+
+def assert_only_direct_relations(*, path: RelationPath):
+    """Ensure that only direct relations are supported which should be the case if the
+    relation path only contains one path element.
+    """
+    num_elements = len(path.elements)
+    if num_elements != 1:
+        raise ModelAssumptionError(
+            f"The provided relation path {
+                path.path_str}"
+            f"does not describe a direct relation, but contains {
+                num_elements} different relations"
+        )
+
+
+def assert_class_is_source(*, path: RelationPath, instruction: InstructionProtocol):
+    """Ensure that the class being modified with the reference count property is the expected class.
+    This function iterates over the elements of the relation path in the given instruction
+    and validates that the class being modified with the reference count property matches
+    the class specified in the relation path.
```

Comment:
Second part of the sentence seems to be redundant with the initial description in the first line.

Reply:
Also, reference to `reference count` property should be removed.